client/src/components/Header/components/NavLinks.js

```diff
@@ -1,29 +1,39 @@
 import React from 'react';
 import { Link } from '../../helpers';
-
 import PropTypes from 'prop-types';
 
+import Login from '../components/Login';
+import { avatarRenderer } from '../../../utils/avatarRenderer';
+
 const propTypes = {
-  displayMenu: PropTypes.bool
+  displayMenu: PropTypes.bool,
+  pathName: PropTypes.string.isRequired,
+  user: PropTypes.object
 };
 
-function NavLinks({ displayMenu }) {
+export function authOrUserLink(user, pathName) {
+  const isUserDonating = user && user.isDonating;
+  const isUserSignedIn = user && user.username;
+
+  if (pathName === '/' || !isUserSignedIn) {
+    return <Login data-test-label='landing-small-cta'> Sign In </Login>;
+  } else if (isUserDonating) {
+    return (
+      <Link className='supporter-nav-avatar' to={`/${user.username}`}>
+        {avatarRenderer(user.picture, user.username, 34, 1)}
```

Comment:
```js
        {avatarRenderer(user.picture, user.username, 34, 1)}
```

Not a fan of the magic numbers here. We should be able to move this away into the component. This is clearly not dynamic because we are hard coding it. Does this need to be passed as a prop? Is there are better cleaner way.

Reply:
I could move them to states, but it would not make any difference since the height of the nav is fixed and these numbers will not change.

This a similar convention to what we have here for rendering camper avatar.
 client/src/components/profile/components/Camper.js

Reply:
> these numbers will not change.

Exactly my point.

Since these values are fixed, it implies that they need to be set on a particular condition? This is an imperative approach and it could pretty straight to make it declarative. 

You already know what you want when the component needs to be placed in the nav, its needs to be 34px, with a padding of 1px it could be something else on a different component like on the profile page 256px with a padding of 5px.

Passing the css properties which are internal and native to the avatar component is imperative and breaks the notion of writing it as a component in the first place. It should be able to return the correct version of the component when asked by the nav or the profile page declaratively.

Reply:
Just to clarify, I would need to pass "profile" and "navigation" to avatar renderer and handle the sizes there?

Reply:
Is it okay if I push a change? I can demonstrate better with code.

Reply:
of course.

Reply:
@raisedadead, I am not sure I was able to understand your previous comment. If you tell me about the specific implementation you have in mind, I can make the changes accordingly. Thank you.

Reply:
Yikes, I forgot about this. 

We should do this declaratively, i.e something along the lines of

```js
{avatarRenderer(user.picture, user.username, 'sm')}
{avatarRenderer(user.picture, user.username, 'lg')}
```

the last param can be defined as `size` for example.

where the size='sm' can be equated to 34px width, 34px height, 1px padding
and the size='md' can be equated to 100px width, 100px height, 2px padding
and the size='lg' can be equated to 256px width, 256px height, 5px padding

This would make the component declative, with below uses. 

1. You can change the component later to whatever other internal styling without needing to change code all over the place.
2. You will not break the api of the `avatarRenderer`, should you forget to pass a param
3. You will be able to use the component elsewhere in pages if needed.

Basically you will be writing this as a 'component' vs 'implementation', the way it is right now makes it neither. There is a component but we are passing its implementation values.

Reply:
Could we make `avatarRenderer` a standard component?  Rather than passing in arguments, we just use props?  So

```
{avatarRenderer(user.picture, user.username, 'sm')}
```
becomes

```
<Avatar picture={user.picture} username={user.picture} size={'sm'}/>
```

or 

```
<Avatar user={user} size={'sm'}/>
```

and we `isRequired` the `picture` and `username` properties inside `Avatar`.  I think this is a more idiomatic way of going about it.

Reply:
Yup it should be a standard component.